Mathlib/RingTheory/Coalgebra/Equiv.lean

```diff
@@ -246,6 +246,42 @@ theorem trans_toCoalgHom :
 theorem coe_toEquiv_trans : (e₁₂ : A ≃ B).trans e₂₃ = (e₁₂.trans e₂₃ : A ≃ C) :=
   rfl
 
+/-- If an coalgebra morphism has an inverse, it is an coalgebra isomorphism. -/
+@[simps]
+def ofCoalgHom (f : A →ₗc[R] B) (g : B →ₗc[R] A) (h₁ : f.comp g = CoalgHom.id R B)
+    (h₂ : g.comp f = CoalgHom.id R A) : A ≃ₗc[R] B :=
+  { f with
+    toFun := f
+    invFun := g
+    left_inv := CoalgHom.ext_iff.1 h₂
+    right_inv := CoalgHom.ext_iff.1 h₁ }
```

Comment:
```suggestion
@[simps]
def ofCoalgHom (f : A →ₗc[R] B) (g : B →ₗc[R] A) (h₁ : f.comp g = CoalgHom.id R B)
    (h₂ : g.comp f = CoalgHom.id R A) : A ≃ₗc[R] B where
  __ := f
  toFun := f
  invFun := g
  left_inv := CoalgHom.ext_iff.1 h₂
  right_inv := CoalgHom.ext_iff.1 h₁
```

Reply:
Note that the `simps` line is generating results like
```lean
CoalgEquiv.ofCoalgHom_apply.{u_1, u_2, u_3} {R : Type u_1} {A : Type u_2} {B : Type u_3} [CommSemiring R]
  [AddCommMonoid A] [AddCommMonoid B] [Module R A] [Module R B] [CoalgebraStruct R A] [CoalgebraStruct R B]
  (f : A →ₗc[R] B) (g : B →ₗc[R] A) (h₁ : f.comp g = CoalgHom.id R B) (h₂ : g.comp f = CoalgHom.id R A) (a : A) :
  (ofCoalgHom f g h₁ h₂) a = f a
```
that are very close to your `coe_coalgHom_ofCoalgHom`. I am not sure we want both. Anyway, if this is `simps` also `ofBialgHom` should be `simps`.